api/checks/runs.go

```diff
@@ -15,15 +15,28 @@ import (
 
 func updateCheckRunSummary(ctx context.Context, summary summaries.Summary, suites ...shared.CheckSuite) (bool, error) {
 	log := shared.GetLogger(ctx)
+	if len(suites) < 1 {
+		return false, nil
+	}
 	product := summary.GetCheckState().Product
-
 	testRun := summary.GetCheckState().TestRun
+
+	// Attempt to update any existing check runs for this SHA.
+	checkRuns, err := getExistingCheckRuns(ctx, suites[0])
+	if err != nil {
+		log.Warningf("Failed to load existing check runs for %s: %s", suites[0].SHA[:7], err.Error())
+	}
+
+	createdAny := false
+	errors := make(chan error, len(suites))
```

Comment:
I don't see any gorountine here. What's the benefit of using a channel instead of a slice?